mcsema/BC/Function.cpp

```diff
@@ -260,6 +288,132 @@ static llvm::BasicBlock *GetOrCreateBlock(TranslationContext &ctx,
   return block;
 }
 
+static void LiftExceptionFrameLP(TranslationContext &ctx,
+                                 const NativeFunction *cfg_func) {
+
+  if (cfg_func->eh_frame.size() > 0) {
+
+    // The personality function `__gxx_personality_v0` is lifted as global variable. Erase the
+    // variable before declaring it as the function.
+    if (auto llvm_used = gModule->getGlobalVariable("__gxx_personality_v0")) {
+      llvm_used->eraseFromParent();
+    }
+
+    auto personality = gModule->getFunction("__gxx_personality_v0");
+
+    if (personality == nullptr) {
+      personality = llvm::Function::Create(
+          llvm::FunctionType::get(llvm::Type::getInt32Ty(*gContext), true),
+          llvm::Function::ExternalLinkage, "__gxx_personality_v0", gModule);
+    }
+
+    auto lifted_func = gModule->getFunction(cfg_func->lifted_name);
+    lifted_func->addFnAttr(llvm::Attribute::UWTable);
+    lifted_func->addFnAttr(llvm::Attribute::OptimizeNone);
+    lifted_func->removeFnAttr(llvm::Attribute::NoUnwind);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 7)
+    lifted_func->setPersonalityFn(personality);
+#endif
+  }
+
+  for (auto &entry : cfg_func->eh_frame) {
+    if (entry->lp_ea == 0) continue;
+
+    if (entry->action == 0) {
```

Comment:
what does `action == 0` imply?

Reply:
`action == 0` implies it's a cleanup handler. It generates the landing pad accordingly.

Reply:
```c++
if (!entry->action) {
  continue;
}
```

Reply:
Are there a fixed number of actions? That is, can this be more meaningfully represented by an `enum`? Otherwise maybe a name like `action_index` would be more appropriate.